examples/cluster/plot_adjusted_for_chance_measures.py

```diff
@@ -3,135 +3,225 @@
+# -------------------------------------------------------------------------
+#
+# In this section we define a similar function that uses several metrics to
+# score 2 uniformly-distributed random labelings. In this case the number of
+# classes and assigned number of clusters are matched for each possible value in
+# `n_clusters_range`.
+
+
+def uniform_labelings_scores(score_func, n_samples, n_clusters_range, n_runs=5):
+    scores = np.zeros((len(n_clusters_range), n_runs))
 
-# Random labeling with varying n_clusters against ground class labels
-# with fixed number of clusters
+    for i, n_clusters in enumerate(n_clusters_range):
+        for j in range(n_runs):
+            labels_a = random_labels(n_samples=n_samples, n_classes=n_clusters)
+            labels_b = random_labels(n_samples=n_samples, n_classes=n_clusters)
+            scores[i, j] = score_func(labels_a, labels_b)
+    return scores
 
-n_samples = 1000
-n_clusters_range = np.linspace(2, 100, 10).astype(int)
-n_classes = 10
+
+# %%
+# In this case we use `n_samples=100` to show the effect of having a number of
+# clusters similar or equal to the number of samples.
+
+n_samples = 100
+n_clusters_range = np.linspace(2, n_samples, 10).astype(int)
 
 plt.figure(2)
 
 plots = []
 names = []
-for score_func in score_funcs:
+for score_name, score_func in score_funcs:
     print(
         "Computing %s for %d values of n_clusters and n_samples=%d"
-        % (score_func.__name__, len(n_clusters_range), n_samples)
+        % (score_name, len(n_clusters_range), n_samples)
     )
 
     t0 = time()
-    scores = uniform_labelings_scores(
-        score_func, n_samples, n_clusters_range, fixed_n_classes=n_classes
-    )
+    scores = uniform_labelings_scores(score_func, n_samples, n_clusters_range)
     print("done in %0.3fs" % (time() - t0))
     plots.append(
-        plt.errorbar(n_clusters_range, scores.mean(axis=1), scores.std(axis=1))[0]
+        plt.errorbar(
+            n_clusters_range, np.median(scores, axis=1), scores.std(axis=1), alpha=0.8
+        )[0]
     )
-    names.append(score_func.__name__)
+    names.append(score_name)
 
 plt.title(
-    "Clustering measures for random uniform labeling\n"
-    "against reference assignment with %d classes" % n_classes
+    "Clustering measures for 2 random uniform labelings\nwith equal number of clusters"
 )
 plt.xlabel("Number of clusters (Number of samples is fixed to %d)" % n_samples)
 plt.ylabel("Score value")
-plt.ylim(bottom=-0.05, top=1.05)
 plt.legend(plots, names)
+plt.ylim(bottom=-0.05, top=1.05)
 plt.show()
+
+# %%
+# We observe similar results as for the first experiment: adjusted for chance
+# metrics stay constantly near zero while other metrics tend to get larger with
+# finer-grained labelings. The mean V-Measure of random labeling increases
+# significantly as the number of clusters is closer to the total number of
+# samples used to compute the measure.
```

Comment:
We could add that, furthermore, raw mutual information is unbounded above and its scale depends on the dimensions of the clustering problem and cardinality of the ground truth classes.